tests/test_flow/test_persistence_fuzz.py

```diff
@@ -221,14 +255,37 @@ def run(self, n_iterations):
                     returned_value = self.model.build_flow().get(affected_name)
                     expected_value = self.model.expected_entity_value(
                         affected_name)
-                    if make_func_change:
+
+                    # Returned value will be different from expected value if we made
+                    # a functional change and updated entity does not change per run.
+                    # If updated entity does change, the value hash will change and
+                    # everything will recompute.
+                    if make_func_change and is_updated_entity_deterministic:
                         assert returned_value != expected_value
                     else:
                         assert returned_value == expected_value
 
-                    # We should have used the cached value, so no new
-                    # computation should have happened.
-                    assert len(self.model.called_entity_names()) == 0
+                    # If updated entity is deterministic, the entity won't recompute.
```

Comment:
This feels like a lot of text for a single assert statement. The text is also confusing: it doesn't provide any introductory context to orient the reader, and it has references to system state and to specific lines of code in other sections.

The amount of effort required to explain this code makes me wonder if we should try to modify the code to be less subtle. However, if that's not possible and we really do need this much text, we should start with a couple of sentences on what this part of the code is trying to accomplish and why it's so tricky.